### PR TITLE
[WFCORE-5027] Adding a proposal for charset and encoding support for credentials in security realms

### PR DESCRIPTION
**Pull request by SoniaZaldana**
https://issues.redhat.com/browse/WFCORE-5027

### CODE REVIEW

File: elytron/WFCORE-5027-charset-encoding-support.adoc [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+
+
+
+==== FileSystem Realm
+
+The proposed changes for the FileSystem realm include adding the options ``hash-encoding`` and
+``hash-charset`` to the ``set-password`` operation as follows:
+
+
+    /subsystem=elytron/filesystem-realm=fsRealm:set-password(identity=user1, digest={algorithm=digest=md5, password=password123, hash-encoding=hex, hash-charset=UTF-16})
```
Comment (fjuma) [inline elytron/WFCORE-5027-charset-encoding-support.adoc:143]: Just a very minor typo, s/digest=md5/digest-md5.

File: elytron/WFCORE-5027-charset-encoding-support.adoc [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+
+=== Other Interested Projects
+
+N/A
+
+== Requirements
+
+=== Hard Requirements
+
+All realms that support hashed passwords need to be modified: Properties Realm, Filesystem Realm, JDBC Realm, and LDAP Realm.
```
Comment (fjuma) [inline elytron/WFCORE-5027-charset-encoding-support.adoc:95]: Would be good to add links to the corresponding Java classes.

**Comment by fjuma** [comment]
Nice analysis, @SoniaZaldana!

File: elytron/WFCORE-5027-charset-encoding-support.adoc (hunk 1 above)
Comment (SoniaZaldana) [reply elytron/WFCORE-5027-charset-encoding-support.adoc]: fixed!

File: elytron/WFCORE-5027-charset-encoding-support.adoc (hunk 2 above)
Comment (SoniaZaldana) [reply elytron/WFCORE-5027-charset-encoding-support.adoc]: Added!

File: elytron/WFCORE-5027-charset-encoding-support.adoc [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+    plain-text=false, hash-encoding="hex", hash-charset="ISO-8859-5")
+
+
+
+The configuration for the properties realm is similar to the ``UserRolesLoginModule`` in the PicketBox configuration
+since these options are resource-wide i.e. these hashing options will be
+used when verifying every single client-provided password against the credentials stored in the
+security realm, as long as hashing is enabled. The only difference lies in that the Elytron Properties realm
+only allows ClearPasswords or DigestPasswords, as opposed to the ``UserRolesLoginModule`` which supports a wider variety of
+hashing algorithms. Supporting more hashing algorithms is out of the scope of this RFE.
```
Comment (SoniaZaldana) [inline elytron/WFCORE-5027-charset-encoding-support.adoc:135]: @fjuma For now, I have written that supporting all the hashing algorithms that legacy supported is out of the scope of this RFE. Should it be included?

**Review by fjuma — APPROVED** [review]

**Pull request merged**